tests/template_tests/syntax_tests/test_basic.py

```diff
@@ -346,6 +346,61 @@ def test_ignores_strings_that_look_like_format_interpolation(self):
         output = self.engine.render_to_string("tpl-weird-percent")
         self.assertEqual(output, "% %s")
 
+    @setup({"template": "{{ class_var.in_template }} | {{ class_var.render_class }}"})
+    def test_access_class_not_initiated(self):
+        class MyClass(list):
+            # As of Python 3.9 list defines __class_getitem__ which makes it
+            # subscriptable.
+            in_template = "Example property"
+            do_not_call_in_templates = True
+
+            @classmethod
+            def render_class(cls):
+                return "Example method"
+
+        output = self.engine.render_to_string("template", {"class_var": MyClass})
+        self.assertEqual(output, "Example property | Example method")
+
+    @setup({"template": "{{ class_var.in_template }}"})
+    def test_access_class_property_returned_by_callable(self):
```

Comment:
```suggestion
    def test_subscriptable_class_returned_by_callable(self):
```

Reply:
Derived thought: shall we unified these two tests and use subTest instead? ensuring that both class variables and class methods are tested in both scenarios:
```python
    @setup({"template": "{{ class_var.class_variable }} | {{ class_var.class_method }}"})
    def test_subscriptable_class(self):
        class MyClass(list):
            # As of Python 3.9 list defines __class_getitem__ which makes it
            # subscriptable.
            class_variable = "Example property"
            do_not_call_in_templates = True

            @classmethod
            def class_method(cls):
                return "Example method"

        for case in (MyClass, lambda: MyClass):
            with self.subTest(case=case):
                output = self.engine.render_to_string("template", {"class_var": case})
                self.assertEqual(output, "Example property | Example method")
```

Reply:
I created the subtest.